
profiles {

    tower-gcp {

        docker {
            enabled = true
        }

        // https://www.nextflow.io/docs/latest/google.html#cloud-batch
        process {
            executor = 'google-batch'
            container = 'us-east1-docker.pkg.dev/longo-48734/pass/python3:latest'
        }
        //workDir = 'gs://nf-tower-6897ad32-d3d4-4bda-ade1-97d25c0e680c/scratch'

        google {
            project = 'longo-48734'
            zone = 'us-east1'
            // location = 'us-east1'
            // batch.spot = true
        }

    }

    gcp {

        docker {
            enabled = true
        }

        // https://www.nextflow.io/docs/latest/google.html#cloud-batch
        process {
            executor = 'google-batch'
            container = 'us-east1-docker.pkg.dev/longo-48734/pass/python3:latest'
        }
        workDir = 'gs://nf-tower-6897ad32-d3d4-4bda-ade1-97d25c0e680c/scratch'

        google {
            project = 'longo-48734'
            zone = 'us-east1'
            // location = 'us-east1'
            // batch.spot = true
        }

    }

    local {

    }

    docker {

        docker {
            enabled = true
        }

        process {
            container = 'us-east1-docker.pkg.dev/longo-48734/pass/python3:latest'
        }

    }

}

// https://stackoverflow.com/questions/77147223/pipeline-fails-when-using-nextflow-tower-google-cloud-batch/77183005#77183005
// required for google batch
plugins {
    id 'nf-validation'
    id 'nf-google@1.7.3-patch1'
}